### PR TITLE
[1LP][RFR] Fix tenant_crud test

### PR DESCRIPTION
**Pull request by okolisny**
Added support for 5.8

{{ pytest: cfme/tests/cloud/test_tenant.py --use-provider rhos7-ga }}

### CODE REVIEW

**Comment by okolisny** [comment]
I don't know why tests in PRT were not run. Tested on local 5.7 and 5.8

**Comment by mkoura** [comment]
please try to use different provider for PRT. rhos7-ga maybe?

**Comment by okolisny** [comment]
@mkoura Test "Skipped". As I understand it demands some special tag in yaml in order to run it

**Comment by mshriver** [comment]
@okolisny please ensure all commits are signed.

File: cfme/cloud/tenant.py [hunk 1]
```diff
@@ -68,10 +72,10 @@ def wait_for_disappear(self, timeout=300):
         except TimedOutError:
             logger.error('Timed out waiting for tenant to disappear, continuing')
 
-    # def wait_for(self, timeout=600):
-    #     return wait_for(self.exists, fail_condition=False, timeout=timeout,
-    #                     message='Wait for cloud tenant to appear', delay=10,
-    #                     fail_func=sel.refresh)
+    def wait_for_appear(self, timeout=600):
+        return wait_for(self.exists, fail_condition=False, timeout=timeout,
```
Comment (mshriver) [inline cfme/cloud/tenant.py]: **Required**
Drop `fail_condition` and `fail_func` parameters. fail_condition is False by default and self.exists calls navigate, which refreshes the page. This fail_func refresh would be executed right before the navigation that occurs during self.exists anyway.

File: cfme/tests/cloud/test_tenant.py [hunk 2]
```diff
@@ -37,7 +37,7 @@ def test_tenant_crud(tenant):
     assert not tenant.exists()
 
     tenant.create()
-    assert tenant.exists()
+    tenant.wait_for_appear()
```
Comment (mshriver) [inline cfme/tests/cloud/test_tenant.py]: Consider keeping `assert tenant.exists()` after the wait_for_appear.

File: cfme/cloud/tenant.py [hunk 3]
```diff
@@ -17,13 +17,15 @@
...
+from utils import version
 from utils.wait import wait_for, TimedOutError
 
 create_tenant_form = Form(
     fields=[
         ('prov_select', AngularSelect("ems_id")),
         ('name', Input('name')),
-        ('save_button', form_buttons.angular_save),
+        ('save_button', version.pick({version.LOWEST: form_buttons.angular_save,
+                                      '5.8': form_buttons.simple_save})),
```
Comment (mfalesni) [inline cfme/cloud/tenant.py]: remove version.pick, Form knows how to handle dict as version pick itself.

**Comment by okolisny** [comment]
@mshriver I returned fail_func for wait_for, because self.exists does not navigate. As I understand, am_i_here method of TenantAll returns True each time

**Comment by mshriver** [comment]
@okolisny you're correct - the am_i_here will return true.  BUT! :)  The resetter() is executed even when am_i_here() is true, unless `use_resetter=False` is explicitly passed to the navigate_to call. So it'll get refreshed even if its already on that page. Make sense?  Check out here for the navigation cycle:
https://github.com/ManageIQ/integration_tests/blob/master/utils/appliance/implementations/ui.py#L368

File: cfme/cloud/tenant.py [hunk 4]
```diff
@@ -68,10 +72,9 @@ def wait_for_disappear(self, timeout=300):
         except TimedOutError:
             logger.error('Timed out waiting for tenant to disappear, continuing')
 
-    # def wait_for(self, timeout=600):
-    #     return wait_for(self.exists, fail_condition=False, timeout=timeout,
-    #                     message='Wait for cloud tenant to appear', delay=10,
-    #                     fail_func=sel.refresh)
+    def wait_for_appear(self, timeout=600):
+        return wait_for(self.exists, timeout=timeout, message='Wait for cloud tenant to appear',
+                        delay=10, fail_func=sel.refresh)
```
Comment (mshriver) [inline cfme/cloud/tenant.py]: Still convincing you to remove the fail_func call ;)

Comment (okolisny) [inline cfme/cloud/tenant.py]: I understood you, and it must work as you said, but it doesn't. I don't know the reason, but what I mentioned - it doesn't refresh the page within self.exists().
Here is the part of [cfme.log](http://pastebin.test.redhat.com/486907), started from wait_for_appear call, without fail_func param. As I understand, ressetter() for Tenant/Details was not called.

Comment (mshriver) [inline cfme/cloud/tenant.py]: Discussed offline - the 'All' destination needs a resetter method for this to work as I was pretending it did.

**Comment by okolisny** [comment]
@mshriver added resetter for 'All' destination, deleted fail_func param inside wait_for_appear
full [cfme.log](http://pastebin.test.redhat.com/486929) of local re-run

**Comment by mshriver** [comment]
Will merge after commits cleaned up a bit, two unsigned in the history

**Pull request merged**